torch/utils/data/graph.py

```diff
@@ -19,6 +20,11 @@ def _stub_unpickler():
 def _list_connected_datapipes(scan_obj, only_datapipe, cache):
     f = io.BytesIO()
     p = pickle.Pickler(f)  # Not going to work for lambdas, but dill infinite loops on typing and can't be used as is
+    if DILL_AVAILABLE:
+        from dill import Pickler
+        d = Pickler(f)
```

Comment:
When `dill` is imported, it will inject logic to pickler dispatch table. You need to revert it after finish traverse by doing `dill.extend(False)`

Reply:
Done